src/fetchers.ts

```diff
@@ -37,25 +83,36 @@ export async function getPodFetchersByAddressOrEns(identifier: string): Promise<
     address = await multicall.resolveName(Name.name);
   }
 
-  const podId = await Name.getText('podId');
-  if (!podId) throw new Error('No podId on ENS found');
+  let podId = await Name.getText('podId');
+  let controllerAddress;
 
-  // Member token tracks Controller for a given pod ID
-  const memberTokenDeployment = getDeployment('MemberToken', network);
-  const MemberToken = new ethers.Contract(
-    memberTokenDeployment.address,
-    memberTokenDeployment.abi,
-    multicall,
-  );
+  if (podId) {
+    controllerAddress = await getControllerAddressByPodId(podId);
+  } else {
+    // fallback checker for pods without .pod ENS names
+    controllerAddress = await getControllerAddressByPodAddress(address);
```

Comment:
🎨 typically we grab the pod id from an id text field that we add to the .pod.xyz names when we create them. If a pod changes its ens name we can't use that field and have to grab our controller from the safe itself.